drivers/SmartThings/matter-sensor/src/init.lua

```diff
@@ -91,6 +117,28 @@ local function temperature_attr_handler(driver, device, ib, response)
   device:emit_event_for_endpoint(ib.endpoint_id, capabilities.temperatureMeasurement.temperature({value = temp, unit = unit}))
 end
 
+local temp_attr_handler_factory = function(minOrMax)
+  return function(driver, device, ib, response)
+    if ib.data.value == nil then
+      return
+    end
+    local temp = ib.data.value / 100.0
+    local unit = "C"
+    set_field_for_endpoint(device, TEMP_BOUND_RECEIVED..minOrMax, ib.endpoint_id, temp)
+    local min = get_field_for_endpoint(device, TEMP_BOUND_RECEIVED..TEMP_MIN, ib.endpoint_id)
+    local max = get_field_for_endpoint(device, TEMP_BOUND_RECEIVED..TEMP_MAX, ib.endpoint_id)
+    if min ~= nil and max ~= nil then
+      if min < max then
+        device:emit_event_for_endpoint(ib.endpoint_id, capabilities.temperatureMeasurement.temperatureRange({ value = { minimum = min, maximum = max }, unit = unit }))
```

Comment:
The temperature conversions for tempRange (and the setpoint values as well) are supported in FW 53+. Therefore, we won't want to send these requests for cases where the hub is running FW 53 or greater. We could gate this on lua libs api version 10 since that is the api version for lua libs packaged with FW 53.

Caveat: Lua libs version until now has essentially been tied to FW version since we use the lua libs packaged in FW. However, this parity may be broken with async lua libs releases, so this is something we will need to consider. The change to support these temp conversion is not technically in the lua libs, it is in the FW.

This will be the same for the `matter-thermostat` PR as well: https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/pull/1223

Reply:
As discussed, added a check for RPC version in 2afbc40e1f113c7c03763e87d4061378c41ddc1f.